migrator/src/main/java/uk/nhs/digital/ps/migrator/model/hippo/Attachment.java

```diff
@@ -0,0 +1,93 @@
+package uk.nhs.digital.ps.migrator.model.hippo;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import uk.nhs.digital.ps.migrator.PublicationSystemMigrator;
+import uk.nhs.digital.ps.migrator.config.ExecutionParameters;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.net.URL;
+import java.net.URLConnection;
+import java.nio.channels.Channels;
+import java.nio.channels.FileChannel;
+import java.nio.channels.ReadableByteChannel;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class Attachment {
+
+    private static final Logger log = LoggerFactory.getLogger(Attachment.class);
+
+    private static final int DOWNLOAD_CHUNK_SIZE = 1024 * 1024; // 1Mb
+
+    private final String title;
+    private final String uri;
+
+    public Attachment(String title, String uri) {
+        this.title = title;
+        this.uri = uri;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getUri() {
+        return uri;
+    }
+
+    public String getFilePath() {
+        ExecutionParameters executionParameters = PublicationSystemMigrator.getExecutionParameters();
+        return executionParameters.getAttachmentDownloadDir() + getUri();
+    }
+
+    public String getMimeType() {
+        try {
+            return Files.probeContentType(Paths.get(getFilePath()));
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public void downloadAttachment() {
+
+        try {
+            File file = new File(getFilePath());
+            if (file.exists() && file.length() > 0) {
+                log.debug("File {} already exists, not downloading.", getFilePath());
+                return;
+            }
+
+            file.getParentFile().mkdirs();
+            file.createNewFile();
+
+            try (FileOutputStream fos = new FileOutputStream(file)) {
+                // The URIs have spaces and we need to encode them or we get a 505 error response
+                String encodedUri = getUri().replace(" ", "%20");
+                URL url = new URL("https://indicators.hscic.gov.uk" + encodedUri);
```

Comment:
Could possibly replace with static variable at top of page  similar to DOWNLOAD_CHUNK_SIZE?

Reply:
Will do 👍